internal/server/deviceflow.go

```diff
@@ -112,29 +171,10 @@ func (a *API) ApproveDeviceAdd(c *gin.Context, req *api.ApproveDeviceFlowRequest
 		return nil, internal.ErrExpired
 	}
 
-	if dfar.AccessKeyID != 0 {
-		// already approved, do nothing
-		return nil, nil
-	}
-
-	// create access key
-	user := rctx.Authenticated.User
-	accessKey := &models.AccessKey{
-		OrganizationMember:  models.OrganizationMember{OrganizationID: rctx.Authenticated.Organization.ID},
-		IssuedFor:           user.ID,
-		IssuedForName:       user.Name,
-		Name:                "Device " + dfar.DeviceCode,
-		ExpiresAt:           rctx.Authenticated.AccessKey.ExpiresAt,
-		InactivityExtension: 30 * days,
-		InactivityTimeout:   time.Now().UTC().Add(30 * days),
-	}
-
-	_, err = access.CreateAccessKey(c, accessKey)
-	if err != nil {
-		return nil, err
+	if dfar.Approved() {
+		return nil, internal.ErrNotFound
```

Comment:
this removes endpoint idempotency

Reply:
We should probably be idempotent here, but ultimately once the polling code does the login, this record ends up being deleted and the approval will fail. So really there is a <= 5s window where this would be an issue.